receiver/splunkhecrawreceiver/receiver.go

```diff
@@ -161,18 +161,26 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
 	}
 
 	if req.ContentLength == 0 {
-		resp.Write(okRespBody)
+		r.obsrecv.EndLogsOp(ctx, typeStr, 0, nil)
 		return
 	}
 
 	bodyReader := req.Body
 	if encoding == gzipEncoding {
 		var err error
-		bodyReader, err = gzip.NewReader(bodyReader)
+		if r.gzipReader == nil {
+			r.gzipReader, err = gzip.NewReader(bodyReader)
+		} else {
+			err = r.gzipReader.Reset(bodyReader)
```

Comment:
This will only work if you are guaranteed to have non-concurrent HTTP requests, which unless you are doing something special to funnel requests into a single goroutine is _not_ the default behavior in Go's http server.